src/monitors/MonitorOptions.v3

```diff
@@ -57,13 +57,32 @@ component MonitorOptions {
 					.exit<void>();
 			}
 		}
-		var m = MonitorRegistry.lookupName(name);
-		if (m != null) {
+		var entry = MonitorRegistry.lookupName(name);
+		if (entry != null) {
 			if (args != null) {
-				var msg = m.configure(args);
+				parseMonitorArgs(entry, args);
+				var msg = entry.monitor.configure(args);
 				if (msg != null) Trace.OUT.put2("%s: %s", name, msg).ln();
 			}
-			monitors.put(m);
+			monitors.put(entry.monitor);
+		}
+	}
+	private def parseMonitorArgs(entry: MonitorRegistryEntry, args: string) {
+		var s = 0, pos = 0;
+		var arg: string;
+		while (pos < args.length) {
+			match (args[pos]) {
+				',' => {
+					arg = Arrays.prepend('-', Arrays.range(args, s, pos));
+					entry.o.parseOption(arg);
+					s = ++pos;
+				}
+				_ => pos++;
+			}
+		}
+		if (s < args.length) {
+			arg =Arrays.prepend('-', Arrays.range(args, s, args.length));
```

Comment:
@titzer this, along with the changes in MonitorRegistry enables the monitors to effectively use `OptionGroup`. Basically each `MonitoryRegistryEntry` is now an `OptionGroup`, and as long as the options are specified in the individual monitors file, help text is automatically generated, and parsing happens automatically.

I intentionally made parsing happen before calling `configure` so that monitors can implement error handling logic

Reply:
btw the help text looks like this now
<img width="840" alt="Screenshot 2024-09-10 at 8 06 44 PM" src="https://github.com/user-attachments/assets/cf0643b5-7928-49d8-8c3b-e126191dd21f">

Reply:
As discussed in person, just a minor refactoring to use composition rather than inheritance here.